migrate.js

```diff
@@ -91,6 +82,71 @@ function scanAndCount(pushstream, cb) {
   })
 }
 
+/**
+ * Fallback algorithm that does the same as findMigratedOffset, but is slow
+ * because it does a scan of BOTH new log and old log.
+ */
+function inefficientFindMigratedOffset(newLog, oldLog, cb) {
+  scanAndCount(newLog.stream({ gte: 0 }), (err, msgCountNewLog) => {
+    if (err) return cb(err) // TODO: might need an explain() here
+    if (!msgCountNewLog) return cb(null, -1)
+
+    let result = null
+    pull(
+      oldLog.getStream({ gte: 0 }),
+      pull.take(msgCountNewLog),
+      pull.drain(
+        (x) => {
+          result = x.seq
+        },
+        (err) => {
+          if (err) return cb(err) // TODO: might need an explain() here
+          cb(null, result)
+        }
+      )
+    )
+  })
+}
+
+function findMigratedOffset(sbot, oldLog, newLog, cb) {
+  if (!sbot.get) {
+    debug('running in inefficient mode because no ssb-db is installed')
+    inefficientFindMigratedOffset(newLog, oldLog, cb)
+    return
+  }
+
+  newLog.onDrain(() => {
+    if (typeof newLog.since.value !== 'number' || newLog.since.value < 0) {
+      cb(null, -1)
```

Comment:
This `if` condition will be triggered by the new test added. Without that new test, we were potentially exposed to the lack of this `if` condition passing all tests, when in production it could break.